xmtp_id/src/associations/member.rs

```diff
@@ -76,52 +94,244 @@ impl MemberIdentifier {
     /// Get the value for [`MemberIdentifier::Installation`] variant.
     /// Returns `None` if the type is not the correct variant.
     pub fn to_installation(&self) -> Option<&[u8]> {
-        if let Self::Installation(ref installation) = self {
-            Some(installation)
+        if let Self::Installation(installation) = self {
+            Some(&installation.0)
         } else {
             None
         }
     }
 }
 
-impl std::fmt::Display for MemberIdentifier {
-    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        let as_string = match self {
-            MemberIdentifier::Address(address) => address.to_string(),
-            MemberIdentifier::Installation(installation) => hex::encode(installation),
+impl PublicIdentifier {
+    #[cfg(any(test, feature = "test-utils"))]
+    pub fn rand_ethereum() -> Self {
+        Self::Ethereum(ident::Ethereum::rand())
+    }
+
+    pub fn sanitize(self) -> Result<Self, IdentifierValidationError> {
+        let ident = match self {
+            Self::Ethereum(addr) => Self::Ethereum(addr.sanitize()?),
+            ident => ident,
+        };
+        Ok(ident)
+    }
+
+    pub fn eth(addr: impl ToString) -> Result<Self, IdentifierValidationError> {
+        Self::Ethereum(ident::Ethereum(addr.to_string())).sanitize()
+    }
+
+    pub fn passkey(key: Vec<u8>, relying_partner: Option<String>) -> Self {
+        Self::Passkey(ident::Passkey {
+            key,
+            relying_partner,
+        })
+    }
+
+    pub fn passkey_str(
+        key: &str,
+        relying_partner: Option<String>,
+    ) -> Result<Self, IdentifierValidationError> {
+        Ok(Self::Passkey(ident::Passkey {
+            key: hex::decode(key)?,
+            relying_partner,
+        }))
+    }
+
+    pub fn from_proto(
+        ident: impl AsRef<str>,
+        kind: IdentifierKind,
+    ) -> Result<Self, ConversionError> {
+        let ident = ident.as_ref();
+        let public_ident = match kind {
+            IdentifierKind::Unspecified | IdentifierKind::Ethereum => {
```

Comment:
Same question here: wdyt of double checking the address if it's really eth address?